pkg/osvscanner/vulnerability_result.go

```diff
@@ -74,3 +75,58 @@ func groupResponseBySource(r reporter.Reporter, query osv.BatchedQuery, resp *os
 
 	return output
 }
+
+// diffVulnerabilityResults will return any new vulnerabilities that are in `newRes`
+// which is not present in `oldRes`, but not the reverse.
+//
+// Current implementation is O(n^2) on the number of vulns, but can be reduced to linear time
+func diffVulnerabilityResults(oldRes, newRes models.VulnerabilityResults) models.VulnerabilityResults {
+	result := models.VulnerabilityResults{}
+	for _, ps := range newRes.Results {
+		sourceIdx := slices.IndexFunc(oldRes.Results, func(elem models.PackageSource) bool { return elem.Source == ps.Source })
+		if sourceIdx == -1 {
```

Comment:
nit: add more comments for the different cases here.

e.g. here: "Entire source finding is new"  or some variation.

Reply:
Done in: 5f1fd7a471c91a6065a92814389ab6399a399bee